src/commands/serve.rs

```diff
@@ -158,6 +158,28 @@ pub async fn serve(bind_address: &str, port: &str) -> std::io::Result<()> {
         }
     }
 
+    fn get_available_port(port: u16, bind_address: &str) -> Option<u16> {
+        let port_available = |port: u16, bind_address: &str| {
+            std::net::TcpListener::bind((bind_address, port)).is_ok()
```

Comment:
This is technically a race condition. What if the port was available at this point but becomes unavailable at the bind call?

Ideally we should run the bind call in the loop.